src/mercury_engine_data_structures/formats/bmsld.py

```diff
@@ -220,5 +268,52 @@ def compare_func(first: str, second: str) -> bool:
 
     def insert_into_entity_group(self, sub_area: Container, name_to_add: str) -> None:
         # MSR requires to have the names in the sub area list sorted by their crc32 value
-        sub_area.names.append(name_to_add)
-        sub_area.names.sort(key=crc32)
+        sub_area.objects.append(name_to_add)
+        sub_area.objects.sort(key=crc32)
+
+    def get_layer(self, layer: ActorLayer) -> Container:
```

Comment:
technically this returns a ListContainer, although I'm unsure if we should be exposing this at all.

Reply:
if you need it for other functions, prefix it with an underscore (`_get_layer()`) to indicate it's a private function, not part of the public API

Reply:
I have found some uses for it in the patcher, so I'll add the underscore.

Reply:
Oh whoops. Misread.

Reply:
ideally, it shouldn't be used in the patcher. what do you need it for?

Reply:
example: `door_actor = scenario.get_layer(ActorLayer.DOOR).get(door_name, None)`

Reply:
you shouldn't be doing that. `door_actor = scenario.get_actor(ActorLayer.DOOR, door_name)` is the correct way. if you need additional functionality (e.g. optionally returning None if the actor doesn't exist) then add it to the API